### ISSUE TITLE
MQTT unstable

### ISSUE THREAD

**Issue by stheine** [tillsteinbach/VWsFriend#698]
Maybe related to #531 and #685, I see several disconnects per day, all ending with the 'weconnectConnected False'.
I now have an external monitor, starting a timeout, and whenever that situation does not resolve within 5minutes, restart the docker container.

This is based on the `tillsteinbach/vwsfriend:latest` docker image with `WECONNECT_INTERVAL=180`.

**Comment by wormiedk** [comment]
@stheine I also see the same without MQTT enabled. 

How do you monitor it is that via absence of MQTT messages?

**Comment by stheine** [comment]
@wormiedk 

> I also see the same without MQTT enabled.
correct, it's not MQTT that is unstable, but the overall vwsfriend client seems to get stuck regularly.
it's just MQTT that makes me monitor that state.

> How do you monitor it is that via absence of MQTT messages?
I have set up a job that listens to the 'vwsfriend/mqtt/weconnectUpdated' message. when it arrives, it starts a timer for twice the updateInterval to trigger a restart of vwsfriend. and when there is already such a timer, it is cleaned.

t: weconnectUpdated: start timer for 2*interval
t + interval: weconnectUpdated: clean previous timer, start new timer for 2*interval
t + 2*interval: weconnectUpdated: clean previous timer, start new timer for 2*interval
[vwsfriend gets stuck]
t + 4*interval: timer gets triggered and restarts vwsfriend

and that process has worked fine since November.


just recently I have switched to tillsteinbach's new carconnectivity-mqtt client, and that seems to be more stable so far.

**Comment by wormiedk** [comment]
I will try that. At the moment I am using https://github.com/robinostlund/homeassistant-volkswagencarnet for home assistant but am missing the vwsfriend stats... Might need to build something with automations for logging charging
